src/redux/features/globalRoles/globalRolesSlice.ts

```diff
@@ -0,0 +1,39 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+import { StatusT, buildReducers } from 'src/redux/features/asyncUtils';
+import { GlobalRolesUsersData } from 'src/types/GlobalRoles';
+
+import { requestListGlobalRolesUsers, requestUpdateGlobalRolesUser } from './globalRolesAsyncThunks';
+
+/**
+ * Users with global roles list
+ */
+const initialStateGlobalRolesUsersList: { [key: string]: StatusT<GlobalRolesUsersData> } = {};
+
+export const globalRolesUsersListSlice = createSlice({
+  name: 'globalRolesUsersListSlice',
+  initialState: initialStateGlobalRolesUsersList,
+  reducers: {},
+  extraReducers: (builder) => {
+    buildReducers(requestListGlobalRolesUsers)(builder);
```

Comment:
We could probably use a pre-defined key for this, 'list' or something like that. Might need a small tweak in the buildReducers function to accept an optional callback to get the key.

Reply:
I don't follow, what would that be for? Just so we don't have to track the request ID in the consumer?

Reply:
Yes that and keeping one state in the store for list data.. and we can show data that's already in the store before dispatch gets a chance to complete (on repeat visits).
Just something we did before assuming list/search sorta follows a get by id pattern, just that the id here is a predetermined 'list'. 
Sorry about the confusion, this is fine as a current pattern, no need to change.

Reply:
We would potentially show incorrect data prefetched from the store, if the search/sort/filters are different in the current view. Scrap this.

Reply:
Oh I see what you're saying. Yea I think maybe this is where I should be using `createCachedSelector`. I am going to table this work now that the foundation is done since it is going to end up in another PRD. When we come back to it maybe we can implement `re-reselect`, and it seems like we should use it anywhere we are doing locally filtered search results too.